realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -456,6 +456,9 @@ private void emitGetFieldNamesMethod(JavaWriter writer) throws IOException {
     private void emitToStringMethod(JavaWriter writer) throws IOException {
         writer.emitAnnotation("Override");
         writer.beginMethod("String", "toString", EnumSet.of(Modifier.PUBLIC));
+        writer.beginControlFlow("if (!isValid())");
```

Comment:
I noticed that Cocoa did the same and it makes sense instead of throwing RealmException which might be a bit unexpected on a standard java method.